sources/engine/Stride.Assets/SpriteFont/SystemFontProvider.cs

```diff
@@ -52,72 +50,105 @@ public SystemFontProvider(string fontName)
+                var weight = Style.IsBold() ? FontWeight.Bold : FontWeight.Regular;
+                var style = Style.IsItalic() ? SharpDX.DirectWrite.FontStyle.Italic : SharpDX.DirectWrite.FontStyle.Normal;
+                font = fontFamily.GetFirstMatchingFont(weight, FontStretch.Normal, style);
             }
 
             return new FontFace(font);
         }
 
         public override string GetFontPath(AssetCompilerResult result = null)
         {
-            using (var factory = new Factory())
+            if(OperatingSystem.IsWindows())
+                return GetFontPathWindows(result);
+            if(OperatingSystem.IsLinux())
             {
-                Font font;
+                var fontPath = GetFontPathLinux(result);
+                if(fontPath == null && FontName != GetDefaultFontName())
+                {
+                    result?.Warning($"Cannot find font family '{FontName}'. Loading default font '{GetDefaultFontName()}' instead");
+                    FontName = GetDefaultFontName();
+                    fontPath = GetFontPathLinux(result);
+                }
+                return fontPath;
+            }
+            return null;
+        }
+        private string GetFontPathLinux(AssetCompilerResult result)
+        {
+            var bold = Style.IsBold() ? "Bold" : "";
+            var italic = Style.IsItalic() ? "Italic" : "";
+            string systemFontDirectory = "/usr/share/fonts";
+            string[] files = System.IO.Directory.GetFiles(systemFontDirectory, "*.ttf", System.IO.SearchOption.AllDirectories);
 
-                using (var fontCollection = factory.GetSystemFontCollection(false))
+            foreach (string file in files)
+            {
+                if (file.Contains(FontName) && file.Contains(bold) && file.Contains(italic))
                 {
-                    int index;
-                    if (!fontCollection.FindFamilyName(FontName, out index))
-                    {
-                        result?.Error($"Cannot find system font '{FontName}'. Make sure it is installed on this machine.");
-                        return null;
-                    }
-
-                    using (var fontFamily = fontCollection.GetFontFamily(index))
-                    {
-                        var weight = Style.IsBold() ? FontWeight.Bold : FontWeight.Regular;
-                        var style = Style.IsItalic() ? SharpDX.DirectWrite.FontStyle.Italic : SharpDX.DirectWrite.FontStyle.Normal;
-                        font = fontFamily.GetFirstMatchingFont(weight, FontStretch.Normal, style);
-                        if (font == null)
-                        {
-                            result?.Error($"Cannot find style '{Style}' for font family {FontName}. Make sure it is installed on this machine.");
-                            return null;
-                        }
-                    }
+                    return file;
```

Comment:
I'm guessing there are less hacky ways to get the style from a font, probably have to read the .ttf though ? If that's the case, can you add a todo here for this ?

Insert a line break between this function and the previous one.
`GetFiles` is fairly inefficient for this specific use case, best replace it with a `Directory.EnumerateFiles(systemFontDirectory, "*.ttf", System.IO.SearchOption.AllDirectories)` instead.

Reply:
> I'm guessing there are less hacky ways to get the style from a font, probably have to read the .ttf though

Yeah, I'll try to see if it is possible to verify the style with our current FreeType binding, but common convetion I noticed is also separating styles into multiple files. I'm not though sure is always the case - see C:\Windows\Fonts where all fonts have single file for each style.

Perhaps I should also consider adding a specific font to the Search regex. Something like this:
`Directory.EnumerateFiles(systemFontDirectory, $"{FontName}*.ttf", System.IO.SearchOption.AllDirectories)`